[#2] 인증 메일 보내기

ahinkim opened the pull request.

ahinkim committed:
#[2] feat: 인증 메일 보내기
인증 메일(보안 코드) 전송 기능 구현
인증 메일 재전송시 redis key(email)에 대한 value(보안 코드)만 바뀌도록 재전송에 대한 예외 처리 구현
인증 메일 html 생성

ahinkim committed:
[#2] feat: 인증 확인 기능
보안 코드 비교 후 사용자의 이메일이 맞는 지 인증하기
보안 코드가 다를 경우 NotFoundException 발생

ahinkim committed:
[#2] refactor: 인증 메일 기능 개선
URI 규칙에 맞게 수정 - 컨트롤 자원을 의미하는 경우 명사보다 동사 사용
response 형식에 맞게 수정

ahinkim committed:
[#2] fix: 인증 메일 기능 개선
클라이언트 요청, 서버 에러로 발생하는 Exception 처리
상태코드와 메시지가 다르게 전송되는 문제 ResponseEntity 추가 구현으로 개선

ahinkim committed:
[#2] refactor: 인증 메일 기능 개선
이메일 글자 수 제한 기능 추가 구현

ahinkim committed:
[#2] refactor: 인증 메일 기능 개선
Java Coding convention 규칙에 맞게 code style 적용

ahinkim committed:
Merge branch 'Feature/SendEmail' of https://github.com/MEMORY-MOA/BE-MEMBER into Feature/SendEmail

ahinkim committed:
Merge branch 'Feature/SendEmail' of https://github.com/MEMORY-MOA/BE-MEMBER into Feature/SendEmail

yejin013 commented on src/main/java/com/moa/member/service/MemberServiceImpl.java:
```diff
@@ -1,21 +1,115 @@
+		String verificationCode = request.getVerificationCode();
+
+		String storedVerificationCode = redisUtil.getData(email);
+<<<<<<< HEAD
```
주석 덜 지운 것 같아 확인해주세요~

ahinkim replied:
아 아직 어제 얘기했던 건 변경 못했어! 주석 다 지우고 다시 올리겠습니당

yejin013 commented on src/main/java/com/moa/member/service/MemberServiceImpl.java:
```diff
@@ -1,21 +1,115 @@
+		return message;
+	}
+
+	public static String createCode() {
```
이거 private으로 하는 건 어때? 서비스에서만 쓰는거지?

ahinkim replied:
웅웅 서비스에서만! ㅇㅋ

yejin013 commented on src/main/java/com/moa/member/service/MemberServiceImpl.java:
```diff
@@ -1,21 +1,115 @@
+		memberRepository.save(member);
+	}
+
+	private MimeMessage createMessage(String to, String code) throws Exception {
```
단순 질문! 이건 왜 throws Exception 있어? 어떤 것 때문에 넣은거야???? MimeMessageHelper 때문에 무조건 해야하는 건가?

ahinkim replied:
웅웅 무조건 구현해야하는 부분! 예외 명시 안하면 에러나더라구

yejin013 commented on src/main/java/com/moa/member/service/MemberServiceImpl.java:
```diff
@@ -1,21 +1,115 @@
+			emailSender.send(message);
+		} catch (MailException es) {
+			es.printStackTrace();
+			throw new RuntimeException("Failed to send email.");
```
한글로 바꿔주세요~

yejin013 replied:
흠... 그리고 고민되는 게 RuntimeException은 주로 custom exception으로도 많이 하잖아... Runtime으로 둘건지 Custom 만들어서 처리할건지...

ahinkim replied:
> 한글로 바꿔주세요~

아직 어제 얘기한 부분은 반영 안했습니다! 곧 해서 올리겠습니다~

> 흠... 그리고 고민되는 게 RuntimeException은 주로 custom exception으로도 많이 하잖아... Runtime으로 둘건지 Custom 만들어서 처리할건지... 

그런 의미에서는 custom이 맞는 것 같네 이 부분은 더 찾아보고 custom으로 구현하겠습니다

ahinkim committed:
[#2] refactor: 인증 메일 기능 개선
requestDto request 폴더로 분리
불필요한 주석 제거
email 보낼 때 exception 처리 runtime exception에서 custom exception(EmailSendException)으로 수정
application.properties 파일의 보안 설정을 환경 변수 파일(env.properties)로 분리

ahinkim committed:
Merge branch Feature/SendEmail

ahinkim committed:
[#2] refactor: 환경 변수 설정 수정
